feat(plugin-mongodb): TLS configuration support in MongoClientConfig

imsayari404 opened the pull request:
This PR introduces TLS support in the MongoDB connector

## Description
<!---Describe your changes in detail-->

## Motivation and Context
<!---Why is this change required? What problem does it solve?-->
<!---If it fixes an open issue, please link to the issue here.-->

## Impact
<!---Describe any public API or user-facing feature change or any performance impact-->

## Test Plan
```
presto> SELECT * FROM mongodb.tm_lakehouse_engine_db_2.aizen;
 emp_name | emp_location | emp_id | dep_id | u_id 
----------+--------------+--------+--------+------
 wade     | sanfransico  |      7 | AF62   |    2 
 Sam      | dallas       |     46 | Zf6    |    8 
 Root     | kochi        |     12 | AF6    |    4 
 tom      | mumbai       |      4 | Ax5    |    1 
 robert   | hyderabad    |      3 | Zf6    |    9 
 ivan     | bangalore    |     43 | BF4    |   10 
 Yan      | dubai        |      8 | BF4    |    3 
 Tom      | kochi        |     58 | AF6    |    6 
 Jorge    | Sydney       |     34 | Zf6    |    5 
 Bot      | bangalore    |     97 | BF4    |    7 
(10 rows)

Query 20250806_093152_00001_ujc79, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
[Latency: client-side: 0:01, server-side: 0:01] [10 rows, 390B] [8 rows/s, 329B/s]
```

## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

MongoDB Connector Changes
* Add TLS/SSL support with automatic JKS and PEM certificate format detection. Configure using ``mongodb.tls.enabled``, ``mongodb.tls.keystore-path``, 
  ``mongodb.tls.keystore-password``, ``mongodb.tls.truststore-path``, and ``mongodb.tls.truststore-password`` properties.
* Upgrade MongoDB Java Driver to 3.12.14.

```

agrawalreetika reviewed (changes requested):
Please add required documentation entries with the new configs in Mongo connector documentation.

imsayari404 commented on presto-spi/pom.xml:
```diff
@@ -53,6 +53,18 @@
+        <dependency>
+            <groupId>com.facebook.airlift</groupId>
+            <artifactId>security</artifactId>
+            <scope>provided</scope>
```
The presto-spi pom changes are required because I've moved SslContextProvider to the spi package to make it reusable across connectors. The presto-mongodb dependency was added to resolve compilation issues when using SslContextProvider but I have removed them now with the new changes.

imsayari404 replied:
I added the presto-spi changes because  SslContextProvider directly imports and uses Logger from airlift.log and PemReader from airlift.security.pem , without these dependencies, the code won't compile as shown in the compilation errors.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project presto-spi: Compilation failure: Compilation failure: 
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[16,32] package com.facebook.airlift.log does not exist
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[17,41] package com.facebook.airlift.security.pem does not exist
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[54,26] cannot find symbol
[ERROR]   symbol:   class Logger
[ERROR]   location: class com.facebook.presto.spi.security.SslContextProvider
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[54,39] cannot find symbol
[ERROR]   symbol:   variable Logger
[ERROR]   location: class com.facebook.presto.spi.security.SslContextProvider
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[132,24] cannot find symbol
[ERROR]   symbol:   variable PemReader
[ERROR]   location: class com.facebook.presto.spi.security.SslContextProvider
[ERROR] /Users/sayarimukherjee/Desktop/os-presto/presto/presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:[177,54] cannot find symbol
[ERROR]   symbol:   variable PemReader
[ERROR]   location: class com.facebook.presto.spi.security.SslContextProvider
[ERROR] -> [Help 1]
```

imsayari404 commented:
> Does https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/connector/mongodb.rst need any documentation update for this?

@steveburnett , yes the documentation will need to be updated. I'm currently working on addressing the review feedback from @agrawalreetika and will include the necessary documentation updates for all the new TLS configuration properties in the next commit. This will include proper documentation for the new mongodb.tls.* configs and deprecation notices for the old SSL settings. I'll make sure to get these changes pushed soon.

agrawalreetika commented on presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoClientConfig.java:
```diff
@@ -60,7 +63,87 @@ public class MongoClientConfig
+    public boolean isTlsEnabled()
+    {
+        return this.tlsEnabled ||
+                keystorePath != null ||
+                keystorePassword != null ||
+                truststorePath != null ||
+                truststorePassword != null;
```
Why are we checking these here?

imsayari404 replied:
This has been added because what if someone sets mongodb.tls.keystore-path=/path/to/keystore.jks, TLS should be enabled automatically. This will prevent the common configuration mistake where users provide TLS settings but forget to set mongodb.tls.enabled=true. I mean I can remove if you want.

imsayari404 replied:
This was added to handle cases where users provide TLS-related configurations like mongodb.tls.keystore-path but forget to explicitly enable mongodb.tls.enabled=true. Automatically enabling TLS in such cases helps prevent common configuration mistakes. However, I'm happy to remove or adjust this logic if you think it’s unnecessary.

agrawalreetika replied:
But whenever `mongodb.tls.enabled=false` then anyways SSL wouldn't be configured right? https://github.com/prestodb/presto/pull/25374/files#diff-5461194df57bf18a7c80e310cc21d80dc5e078b9707226b0d5f49728c02ae12eR87

imsayari404 replied:
You're  right. I see the issue now. The current logic in isTlsEnabled() is problematic because it automatically enables TLS whenever any TLS-related properties are set, regardless of the mongodb.tls.enabled flag. This could lead to unexpected behavior.
I'll simplify this to just return this.tlsEnabled and remove the automatic detection logic.

agrawalreetika reviewed (commented):
Thanks for making the changes. Overall LGTM.
Please add the documentation & squash all your commits into one.

imsayari404 commented:
> @imsayari404 There are compilation failures, please fix those, add the documentation & squash all your commits into one.

@agrawalreetika The updated changes are now pushed to the branch.

agrawalreetika commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,221 @@
+     */
+    public Optional<SSLContext> buildSslContext()
+    {
+        if (!keystorePath.isPresent() && !truststorePath.isPresent()) {
```
No resolved?

steveburnett commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -46,7 +46,11 @@ Property Name                         Description
+``mongodb.tls.truststore-password``   Password for the truststore file
 ``mongodb.read-preference``           The read preference
 ``mongodb.write-concern``             The write concern
 ``mongodb.required-replica-set``      The required replica set name
```
Yes, I agree! A column for default values will improve the usability of the table.

imsayari404 replied:
Thank you, I will be adding those.

steveburnett reviewed (approved):
LGTM! (docs)

Pull updated branch, new local doc build, looks good. Thank you!

imsayari404 commented:
> LGTM! (docs)
> 
> Pull updated branch, new local doc build, looks good. Thank you!

Appreciate your time and review, @steveburnett, thanks a lot for confirming the doc updates!

steveburnett commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -34,24 +34,28 @@ Configuration Properties
+===================================== ============================================================== ===========
+Property Name                         Description                                                    Default
+===================================== ============================================================== ===========
+``mongodb.seeds``                     List of all mongodb servers
```
```suggestion
``mongodb.seeds``                     List of all MongoDB servers
```

steveburnett commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -34,24 +34,28 @@ Configuration Properties
+===================================== ============================================================== ===========
 
 ``mongodb.seeds``
 ^^^^^^^^^^^^^^^^^
```
Please change "mongod" and "mongos" in line 63 to "MongoDB".

imsayari404 replied:
Sure

steveburnett commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -119,14 +123,45 @@ This flag controls the socket keep alive feature that keeps a connection alive t
+.. note::
+
+    The ``mongodb.ssl.enabled`` property is deprecated and will be removed in a future version.
+    Use ``mongodb.tls.enabled`` instead. The old property name is still supported for backward compatibility.
```
```suggestion
    Use ``mongodb.tls.enabled`` instead. The old property name is supported for backward compatibility.
```

imsayari404 replied:
Thanks, @steveburnett  I've addressed all the suggested changes , corrected the casing of "MongoDB", updated the seed property description, and revised the deprecation note for clarity. Let me know if there's anything else you'd like me to update. Appreciate your review!

steveburnett reviewed (dismissed):
LGTM! (docs) 

Pull updated branch, new local doc build, looks good. 

Thanks for the doc!

pratyakshsharma commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -34,29 +34,33 @@ Configuration Properties
+``mongodb.tls.enabled``               Use TLS/SSL for connections to MongoDB                         ``false``
+``mongodb.tls.keystore-path``         Path to the keystore file for client certificates
+``mongodb.tls.keystore-password``     Password for the keystore file
+``mongodb.tls.truststore-path``       Path to the truststore file for trusted CAs
```
nit: I think it will be good to write the full form rather than CA.

imsayari404 replied:
Sure.. Updated .

pratyakshsharma commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -119,14 +123,45 @@ This flag controls the socket keep alive feature that keeps a connection alive t
+``mongodb.tls.keystore-path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Path to the Java KeyStore file containing the client certificate and private key for TLS authentication. The connector supports both Java KeyStore (JKS) format and PEM file format.
```
nit: add full form for PEM as well

pratyakshsharma commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -119,14 +123,45 @@ This flag controls the socket keep alive feature that keeps a connection alive t
+``mongodb.tls.truststore-path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Path to the Java TrustStore file containing the trusted certificate authorities for TLS connections. The connector supports both Java KeyStore (JKS) format and PEM file format.
```
ditto

pratyakshsharma commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -119,14 +123,45 @@ This flag controls the socket keep alive feature that keeps a connection alive t
+
+Path to the Java TrustStore file containing the trusted certificate authorities for TLS connections. The connector supports both Java KeyStore (JKS) format and PEM file format.
+
+This property is optional and only used when ``mongodb.tls.enabled`` is ``true``. If not specified, the system default truststore will be used.
```
```suggestion
This property is optional and only used when ``mongodb.tls.enabled`` is ``true``. If not specified, the default system truststore will be used.
```

pratyakshsharma commented on presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoClientConfig.java:
```diff
@@ -37,7 +44,11 @@ public void testDefaults()
                 .setSocketTimeout(0)
                 .setSocketKeepAlive(false)
-                .setSslEnabled(false)
+                .setTlsEnabled(false) // Only set TLS, not both SSL and TLS
```
is this comment needed? We have already removed `setSslEnabled` method

pratyakshsharma commented on presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoClientConfig.java:
```diff
@@ -78,7 +96,11 @@ public void testExplicitPropertyMappings()
                 .setSocketTimeout(1)
                 .setSocketKeepAlive(true)
-                .setSslEnabled(true)
+                .setTlsEnabled(true) // Only set TLS, not both
```
ditto.

imsayari404 replied:
Removed

pratyakshsharma commented on presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoClientConfig.java:
```diff
@@ -89,6 +111,27 @@ public void testExplicitPropertyMappings()
+    @Test
+    public void testLegacySslPropertyMapping() throws IOException
+    {
+        Path keystoreFile = Files.createTempFile(null, null);
```
This should create an empty file under /tmp by default. It will be good to clean up these files in finally block?

pratyakshsharma replied:
We can maybe create these files once at class level before any test executes and later clean those files once all tests pass. I see this code is duplicate in all test cases. What do you think?

imsayari404 replied:
I've refactored the code to create the keystore and truststore files once at the class level using @BeforeClass, and cleaned them up in @AfterClass. Removed the duplicate createTempFile calls from individual test methods.

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,221 @@
+    private final Optional<String> keystorePassword;
+    private final Optional<File> truststorePath;
+    private final Optional<String> truststorePassword;
+    private final StandardErrorCode errorCode;
```
Is this really needed? All the connectors using this class can simply throw the same error code and we are currently passing same error code from both places it is called from.

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,221 @@
+            // Fall back to standard KeyStore format
+        }
+
+        try (InputStream inputStream = new FileInputStream(trustStorePath)) {
```
nit: Move above lines in catch block above to be in sync with `loadKeyStore()` method.

imsayari404 replied:
Done

pratyakshsharma reviewed (changes requested):
Thank you for the PR. Few minor comments.

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,217 @@
+            // Attempt to read the key store as a PEM file
+            keystore = PemReader.loadKeyStore(keystoreFile, keystoreFile, keystorePassword);
+            // For PEM encoded keys, the password is used to decrypt the specific key
+            keyManagerPassword = new char[0];
```
This initialisation can be done at the time of declaring.

imsayari404 replied:
Updated

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,217 @@
+        try {
+            // Attempt to read the key store as a PEM file
+            keystore = PemReader.loadKeyStore(keystoreFile, keystoreFile, keystorePassword);
+            // For PEM encoded keys, the password is used to decrypt the specific key
```
Is this the right place to add this comment? This should probably combine with the one on line 128?

imsayari404 replied:
I have made the changes

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,217 @@
+            keyManagerPassword = keystorePassword.map(String::toCharArray).orElse(null);
+            keystore = getInstance(getDefaultType());
+            try (InputStream in = new FileInputStream(keystoreFile)) {
+                keystore.load(in, keyManagerPassword);
```
```suggestion
                keystore.load(in, keystorePassword.map(String::toCharArray).orElse(null));
```

pratyakshsharma replied:
above lines of code can be removed.

pratyakshsharma reviewed (commented):
Final set of comments. Thank you for addressing all earlier comments.

pratyakshsharma reviewed (changes requested):
Couple of high level comments - 
1. Please delete SslContextProvider file from presto-cassandra module since it is no longer getting used.
2. Please change commit message to something like - `Add TLS support in mongodb connector`

imsayari404 commented on presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoClientConfig.java:
```diff
@@ -60,7 +63,83 @@ public class MongoClientConfig
+    @AssertTrue(message = "'mongodb.tls.keystore-path', 'mongodb.tls.keystore-password', 'mongodb.tls.truststore-path' and 'mongodb.tls.truststore-password' must be empty when TLS is disabled")
+    public boolean isValidTlsConfig()
+    {
+        if (!tlsEnabled) {
```
Added

imsayari404 commented:
> Couple of high level comments -
> 
> 1. Please delete SslContextProvider file from presto-cassandra module since it is no longer getting used.
> 2. Please change commit message to something like - `Add TLS support in mongodb connector`

@pratyakshsharma Thank you for your review. I've addressed all your review comments. Could you please re-review when you have a chance?

imsayari404 commented on presto-docs/src/main/sphinx/connector/mongodb.rst:
```diff
@@ -119,14 +124,45 @@ This flag controls the socket keep alive feature that keeps a connection alive t
+Path to the Java KeyStore file containing the client certificate and private key for TLS authentication. The connector supports both Java KeyStore (JKS) format and Privacy-Enhanced Mail (PEM) file format.
 
-This flag enables SSL connections to MongoDB servers.
+This property is optional and only used when ``mongodb.tls.enabled`` is ``true``. Unlike the truststore, there is no default keystore - you must provide one if client certificate authentication is required.
```
Thanks @steveburnett ,  I was trying to make that distinction clear

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+                try (InputStream inputStream = new FileInputStream(trustStorePath)) {
+                    trustStore.load(inputStream, trustStorePassword.map(String::toCharArray).orElse(null));
+                }
+                loaded = true;
```
setting this to true here is redundant, it is not getting used further. Lets remove.

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+    }
+
+    private static KeyStore loadTrustStore(File trustStorePath, Optional<String> trustStorePassword)
+            throws IOException, GeneralSecurityException
```
IOException is not getting thrown and is no more a checked exception for this method. Lets remove it from method signature.

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+            log.debug("Keystore loaded successfully");
+        }
+
+        // Load TrustStore if configured, otherwise use system default (null)
```
I think above comment is misleading. As I understand this, trustStore is getting initialised to system default under `loadTrustStore()` method with the line 
```
KeyStore trustStore = getInstance(getDefaultType());
```

Above comment suggests that there is no system default trustStore and hence we initialise it to null. Please correct me if I am wrong.

imsayari404 replied:
Your understanding is correct:

When a custom truststore path is not provided, the connector does not create a KeyStore instance. It simply passes null to TrustManagerFactory.init(), which causes the JVM to use the system default truststore.

When a custom truststore path is provided, we initialize a KeyStore and populate it with provided certificates.

The misleading part is the comment suggesting a truststore is initialized to null as "system default."
I've updated the comments properly

pratyakshsharma commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+
+        // If no custom truststore is provided, use system default (pass null)
+        // If a custom truststore is provided but empty, that's an error
+        if (truststore != null) {
```
Is this check and above comments needed? Before calling `createTrustManager` method, trustStore is initialised with system default already as part of loadTrustStore() method that I pointed out in my earlier comment. Are we saying system default trustStore can be null in some situations or are we simply putting this check just to be on the safer side?
I find above comment misleading with `use system default (pass null)`. Please help clarify.

imsayari404 replied:
The check for truststore != null is only necessary when a custom truststore is used, to validate its contents.

When no custom truststore is provided, the code passes null to use system defaults, and we do not need to check for empty contents, since the JVM guarantees there’s a default set of trusted certs.

pratyakshsharma reviewed (changes requested):
Few comments around trustStore initialisation again. Please help clarify

pratyakshsharma reviewed (dismissed):
Thank you for addressing all comments. Changes look good to me.

imsayari404 commented:
> Please check why test other modules check is failing. Thanks.

Thank you Pratyaksh, sure.

agrawalreetika commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+            log.debug("Truststore loaded successfully");
+        }
+        else {
+            log.debug("No custom truststore configured, using system default");
```
Wouldn't this cause trust validation to fail for the existing setup if the truststore defaults to JVM defaults instead of the keystore now, when the truststore is missing? 

Since earlier code default to `keystore` -
```
            // load TrustStore if configured, otherwise use KeyStore
            KeyStore truststore = keystore;
            if (truststorePath.isPresent()) {
                truststore = loadTrustStore(truststorePath.get(), truststorePassword);
            }
```

imsayari404 replied:
My main motive behind the change was that there is no default keystore equivalent to the truststore. Unlike the truststore, which can fall back to the system’s default CAs, the keystore contains client‑specific certificates for authentication, so there cannot be a default. Therefore, if no custom keystore is provided, I set it to null instead of being used as a fallback truststore.
That said, I’d like to hear your opinion on this approach.

imsayari404 replied:
Sure, thanks, Reetika, for your suggestion. Your approach makes sense for maintaining backwards compatibility, and I’ll update the code accordingly.

agrawalreetika commented on presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java:
```diff
@@ -0,0 +1,284 @@
+
+    private KeyManager[] createKeyManagers(KeyStore keystore) throws GeneralSecurityException
+    {
+        char[] keyManagerPassword = keystorePassword.map(String::toCharArray).orElse(new char[0]);
```
Also here wouln't this cause problem, earlier the default password was `null` instead?

Earlier it had - 
```
keyManagerPassword = keystorePassword.map(String::toCharArray).orElse(null);
```

agrawalreetika replied:
As per my understanding null may not be safer to APIs that don’t accept null can cause NullPointerException, but implementation wise null and  empty char[] both will have different compatibility issues. And I think An empty char[] isn’t inherently more secure than null — it’s simply a different way of representing a missing value.

IMO, since `null` has been there for earlier implementation, until we have a way to support backward compatibility we can keep it as earlier.

imsayari404 replied:
Sure, will revert the change.

agrawalreetika reviewed (approved):
Thanks, LGTM

imsayari404 commented on presto-plugin-toolkit/pom.xml:
```diff
@@ -125,5 +125,11 @@
             <groupId>com.facebook.airlift</groupId>
             <artifactId>security</artifactId>
         </dependency>
+        <dependency>
```
Thanks for pointing it out. I had added JUnit temporarily while experimenting. I’ll remove it right away and stick with TestNG.

imjalpreet commented on presto-plugin-toolkit/src/test/resources/future-keystore.jks:
or we could generate it programmatically? I just remembered we were trying it for HMS SSL tests: https://github.com/prestodb/presto/pull/25313. Looks like I forgot to review it further but that could be one alternative.

imsayari404 replied:
Updated to use programmatic certificate generation similar to the Hive SSL tests (#25313).

imsayari404 commented:
@tdcmeehan, could you please review this PR at your convenience?

imsayari404 commented on presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoTlsConfiguration.java:
```diff
@@ -0,0 +1,353 @@
+        SslKeystoreManager.initializeKeystoreAndTruststore();
+
+        keystoreFile = new File(SslKeystoreManager.getKeystorePath());
+        truststoreFile = new File(SslKeystoreManager.getTruststorePath());
```
I've updated

agrawalreetika commented on presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoClientModule.java:
```diff
@@ -28,6 +30,8 @@
 public class MongoClientModule
         implements Module
 {
+    private static final Logger log = Logger.get(MongoClientModule.class);
```
unused?

imsayari404 replied:
I've removed

agrawalreetika reviewed (commented):
Mostly lgtm, just few nits
